### PR TITLE
add locations to challenge create page

### PR DESCRIPTION
**Pull request by pbochis**

### CODE REVIEW

File: app/elements/common/location/location-finder.html [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+			#map {
+				height: 100%;
+			}
+			.controls {
+				background-color: #fff;
+				border-radius: 2px;
+				border: 1px solid transparent;
+				box-shadow: 0 2px 6px rgba(0, 0, 0, 0.3);
+				box-sizing: border-box;
+				font-family: Roboto;
```
Comment (pbochis) [inline app/elements/common/location/location-finder.html]: Monica will be sad

File: app/elements/common/location/location-finder.html [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+				}
+			},
+			observers: [
+				'_recenterMap(center)'
+			],
+			_recenterMap: function(){
+				if (!!this.map){
+					this.map.setCenter(this.center);
+				} else {
+					setTimeout(this.initMap(), 1);
```
Comment (lorenzleutgeb) [inline app/elements/common/location/location-finder.html]: Why a timeout of 1ms, and not calling it directly? Furthermore `this.initMap()` will be evaluated and it's return value passed on to `setTimeout` which is probably not what you want. You can wrap that call inside an anonymous function.

File: app/elements/common/location/location-finder.html [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+					// Set the position of the marker using the place ID and location.
+					marker.setPlace({
+						placeId: place.place_id,
+						location: place.geometry.location
+					});
+					marker.setVisible(true);
+
+					infowindow.setContent('<div><strong>' + place.name + '</strong><br>' +
+						'Place ID: ' + place.place_id + '<br>' +
```
Comment (lorenzleutgeb) [inline app/elements/common/location/location-finder.html]: Not sure if we really need to show the Place ID to the user.

File: app/index.html [hunk 4]
```diff
@@ -48,6 +48,7 @@
 	<app-element id="app"></app-element>
 	<!-- build:js scripts/util.js -->
 	<script src="scripts/util.js"></script>
+	<script src="https://maps.googleapis.com/maps/api/js?libraries=places"></script>
```
Comment (lorenzleutgeb) [inline app/index.html]: We should serve that ourselves. Find some bower package that contains it.

File: app/elements/common/location/location-finder.html (hunk 1 above)
Comment (lorenzleutgeb) [inline app/elements/common/location/location-finder.html]: Just no Roboto.

Comment (lorenzleutgeb) [inline app/elements/common/location/location-finder.html]: Or you can add Roboto and a few alternatives if you want.

**Comment by victorbalan** [comment]
i don't understand why we don't use [google-map-search](https://elements.polymer-project.org/elements/google-map?active=google-map-search)

File: app/elements/common/location/google-location-finder.html [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+				},
+				visible: {
+					type: Boolean,
+					value: true
+				},
+				apiConfig: {
+					type : Object,
+					notify: true,
+					value: {
+//							uncomment this and add the api key before building and pushing to prod
```
Comment (lorenzleutgeb) [inline app/elements/common/location/google-location-finder.html]: Nope.

File: app/index.html [hunk 6]
```diff
@@ -48,6 +48,7 @@
 	<app-element id="app"></app-element>
 	<!-- build:js scripts/util.js -->
 	<script src="scripts/util.js"></script>
+	<!--<script src="https://maps.googleapis.com/maps/api/js?libraries=places"></script>-->
```
Comment (lorenzleutgeb) [inline app/index.html]: Do we need this `<script>` now or not?!

**Pull request merged**